### PR TITLE
finished day08

### PR DESCRIPTION
**Pull request by KoenvdBerg**
Hi JP,

Another interesting puzzle. I've written down my approach in the text at the top of the `.scala` file. Wondering what you think of my approach and if you could seen any improvements. Both answers run in about ~25ms on my machine. If your solution is faster I'm curious how you've managed that :). 

Thanks for taking a look! 

- Koen

### CODE REVIEW

File: aoc2022/src/main/scala/day08.scala [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      else false
+
+    if i == 0 || i == trees.length then true                  // tree is on border and thus visible
+    else if j == 0 || j == treesTransposed.length then true   // tree is on border and thus visible
+    else checkRow(i, trees) || checkRow(j, treesTransposed)   // check if tree is taller than surrounding trees
+
+  private val transposedInput = input.transpose
+  private val indices: IndexedSeq[(Int, Int)] = for {
+    i <- Range(0, input.head.length)
+    j <- Range(0, input.length)
```
Comment (jan-pieter) [inline aoc2022/src/main/scala/day08.scala:51]: Alternatively you can write `0 until input.length` which I find more readable.

File: aoc2022/src/main/scala/day08.scala [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+  def isVisible(i: Int, j: Int, trees: TreeRow, treesTransposed: TreeRow): Boolean =
+    def checkRow(index: Int, row: TreeRow): Boolean =
+      val (left, right): (TreeRow, TreeRow) = row.splitAt(index)
+      if left.forall(_ < row(index)) || right.drop(1).forall(_ < row(index)) then true
+      else false
+
+    if i == 0 || i == trees.length then true                  // tree is on border and thus visible
+    else if j == 0 || j == treesTransposed.length then true   // tree is on border and thus visible
+    else checkRow(i, trees) || checkRow(j, treesTransposed)   // check if tree is taller than surrounding trees
```
Comment (jan-pieter) [inline aoc2022/src/main/scala/day08.scala:46]: You can drop the `if/else if/else`

File: aoc2022/src/main/scala/day08.scala [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  type TreeRow = Vector[Int]
+
+  def isVisible(i: Int, j: Int, trees: TreeRow, treesTransposed: TreeRow): Boolean =
+    def checkRow(index: Int, row: TreeRow): Boolean =
+      val (left, right): (TreeRow, TreeRow) = row.splitAt(index)
+      if left.forall(_ < row(index)) || right.drop(1).forall(_ < row(index)) then true
+      else false
+
+    if i == 0 || i == trees.length then true                  // tree is on border and thus visible
+    else if j == 0 || j == treesTransposed.length then true   // tree is on border and thus visible
```
Comment (jan-pieter) [inline aoc2022/src/main/scala/day08.scala:45]: Could there be an off-by-one error here? If I understand correctly `i` and `j` are indices and are always strictly less than the length...

**Comment by KoenvdBerg** [comment]
Hi JP, 

I improved based on your comments. If you're okay with the improvements could you approve?

**Review by jan-pieter — APPROVED** [review]

**Pull request merged**